ols/app/models/config.py

```diff
@@ -383,11 +380,59 @@ def validate_yaml(self) -> None:
         self.logging_config.validate_yaml()
 
 
+class DevConfig(BaseModel):
+    """Developer-mode-only configuration options."""
+
+    enable_dev_ui: bool = False
+    disable_question_validation: bool = False
+    llm_temperature_override: float | None = None
+
+    # TODO - wire this up once auth is implemented
+    disable_auth: bool = False
+
+    def __init__(self, data: Optional[dict] = None):
+        """Initialize developer configuration settings."""
+        super().__init__()
+        if data is None:
+            return
+        self.enable_dev_ui = str(data.get("enable_dev_ui", "False")).lower() == "true"
+        self.disable_question_validation = (
+            str(data.get("disable_question_validation", "False")).lower() == "true"
+        )
+        self.llm_temperature_override = data.get("llm_temperature_override", None)
+        self.disable_auth = str(data.get("disable_auth", "False")).lower() == "true"
+
+    def __eq__(self, other):
+        """Compare two objects for equality."""
+        if isinstance(other, DevConfig):
+            return (
+                self.enable_dev_ui == other.enable_dev_ui
+                and self.disable_question_validation
+                == other.disable_question_validation
+                and self.llm_temperature_override == other.llm_temperature_override
+                and self.disable_auth == other.disable_auth
+            )
+        return False
+
+    def validate_yaml(self) -> None:
+        """Validate OLS Dev config."""
+        if self.llm_temperature_override is not None:
+            if not isinstance(self.llm_temperature_override, (float, int)):
+                raise InvalidConfigurationError(
+                    "llm_temperature_override must be a float"
+                )
+            if self.llm_temperature_override < 0 or self.llm_temperature_override > 1:
+                raise InvalidConfigurationError(
+                    "llm_temperature_override must be between 0 and 100"
```

Comment:
between 0 and 1?

Reply:
the knock on effects from one change......

fixed.